components/common/FormFields.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { ChangeEventHandler } from "react"
+import {
+    Button,
+    Checkbox,
+    Flex,
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Input,
+} from "@chakra-ui/react"
+import styled from "@emotion/styled"
+import ErrorAlert from "./ErrorAlert"
+
+type FormInputFieldProps = {
+    label: string
+    placeholder: string
+    isRequired: boolean
```

Comment:
It would probably be simpler to set
```typescript
isRequired?: boolean
```
and simply leave it undefined if a component is not required.